cabal-install/Distribution/Client/ProjectPlanning.hs

```diff
@@ -1301,6 +1310,14 @@ elaborateInstallPlan platform compiler compilerprogdb
         pkgLibDependencies  = deps
         pkgExeDependencies  = fmap (concatMap (elaborateExeSolverId mapDep)) exe_deps0
         pkgExeDependencyPaths = fmap (concatMap (elaborateExePath mapDep)) exe_deps0
+        pkgPkgConfigDependencies =
+              ordNub
+            $ [ (pn, fromMaybe (error $ "pkgPkgConfigDependencies: impossible! "
+                                          ++ display pn ++ " from " ++ display pkgid)
+                               (pkgConfigDbPkgVersion pkgConfigDB pn))
```

Comment:
Minor suggestion: we use this bulky construction twice in the same function, could pull it into a where clause, e.g. `pkgConfigDbPkgVersion`DammitThisDoesn'tFail or the like